articles/quickstart/native/android/04-user-profile.md

```diff
@@ -110,48 +114,51 @@ profile.getEmail();
 profile.getPictureURL();
 ```
 
-::: note
-Remember that you can't modify the UI inside the onSuccess() method, as it works in a second thread. To solve this, you can persist the data, create a task in the UI thread or create a handler to receive that information.
+::: panel Modifying the UI
+You cannot modify the UI inside the `onSuccess()` method because the method works in a second thread. To solve this issue, you can choose between three options:
+* Persist the data
+* Create a task in the UI thread
+* Create a handler to receive the information
 :::
 
-#### I. ADDITIONAL INFO
+### Additional information
 
-Besides the defaults, you can handle more information that is contained within any of the following `map`:
+You can handle additional information within any of the following maps:
 
-##### A. USER METADATA
+#### A. User metadata
 
-The `userMetadata` map contains fields related to the user profile that can be added from the client-side (e.g. when editing the profile). This tutorial explains how to achieve this. You can access its fields as follows:
+The `userMetadata` map contains user profile fields that can be added from the client-side session (e.g. when a user edits their profile). You can access this information in the following way:
 
 ```java
 String country = (String) profile.getUserMetadata().get("country");
 ```
 
 ::: note
-The strings you use for subscripting the user_metadata map, and the variable types you handle, are up to you.
+You can choose the names and value types of the keys you use for subscripting the `user_metadata` map.
```

Comment:
@lbalmaceda Can you take a look at the revised note? Not sure if I correctly understood what you meant

Reply:
A map/dictionary has a `KEY` and a `VALUE`. The Key is always a String (this is how you want to call it) and the Value can be of many types like a String, Integer, Double, Boolean, Object. i.e.:

```json
profile: {
  "user_metadata":{
    "favorite_quote": "Light in Nature creates the movement of colors.",
    "siblings": 2,
    "money_in_bank": 999.34,
    "is_alive": true,
    "car": {
      "color": "black",
      "wheels": 4,
      "brand": "toyota"
    }
  },
  //More Stuff
}
```

The `user_metadata` is itself a Key in the user's profile, that has a nested Object (more key-values) as the Value. What we want to tell in the above note is that the user can customize the `user_metadata` contents however they want.